src/rust/src/declarative_asn1/decode.rs

```diff
@@ -0,0 +1,55 @@
+// This file is dual licensed under the terms of the Apache License, Version
+// 2.0, and the BSD License. See the LICENSE file in the root of this repository
+// for complete details.
+
+use asn1::{Asn1Readable, Parser};
+use pyo3::types::PyAnyMethods;
+
+use crate::asn1::big_byte_slice_to_py_int;
+use crate::declarative_asn1::types::{AnnotatedType, Type};
+use crate::error::CryptographyError;
+
+type ParseResult<T> = Result<T, CryptographyError>;
+
+fn decode_pyint<'a>(
+    py: pyo3::Python<'a>,
+    parser: &mut Parser<'a>,
+) -> ParseResult<pyo3::Bound<'a, pyo3::types::PyInt>> {
+    let value = asn1::BigInt::parse(parser)?;
```

Comment:
This should be `parser.read_element::<asn1::BigInt>()?`

Reply:
fixed